### DISCUSSION TITLE
Separating Container's using Stack

### DISCUSSION THREAD

**Discussion by runoncedev** [chakra-ui/chakra-ui#5361] [Q&A] [answered]
I'm trying to have 3 stacked containers using `<Stack />` and `<Container />`.

```
import { Stack, Container } from "@chakra-ui/react";

export default function Example() {
  return (
    <Stack>
      <Container maxW="container.sm" border="1px solid">
        Foo
      </Container>
      <Container maxW="container.md" border="1px solid">
        Bar
      </Container>
      <Container maxW="container.sm" border="1px solid">
        Baz
      </Container>
    </Stack>
  );
}
```

The idea of that the middle container is wider than the top and bottom containers and I want to control the spacing between containers using `<Stack />`, but for some reaser after the first container, all other containers end up left-aligned.

Is this approach right? Any suggestions?

You can try it on CodeSandbox: https://codesandbox.io/s/chakra-ui-stack-and-container-bqkrz?file=/src/App.tsx:0-397

Thank you for your hard work!

**Comment by primos63** [comment, marked as answer]
A `Container` has a default style `mx: 'auto'` which makes it centered. `Stack` uses this selector "&>*:not(style) ~ *:not(style)" to apply styles to the siblings of the first `Container`. The style applied depends on the `direction`
```javascript
  const directionStyles = {
    column: {
      marginTop: spacing,
      marginEnd: 0,
      marginBottom: 0,
      marginStart: 0,
    },
    row: { marginTop: 0, marginEnd: 0, marginBottom: 0, marginStart: spacing },
    "column-reverse": {
      marginTop: 0,
      marginEnd: 0,
      marginBottom: spacing,
      marginStart: 0,
    },
    "row-reverse": {
      marginTop: 0,
      marginEnd: spacing,
      marginBottom: 0,
      marginStart: 0,
    },
  }
  ```
  `marginStart` and `marginEnd` correspond to the `mx` attribute. So the two sibling elements have `mx: 'auto'` replaced with `mx: 0`. This is why the second and third containers are left aligned. To overcome this, you can set the `alignItems` attribute to `center`:
  `<Stack alignItems='center'>...</Stack>`.
  
It would be better to have `Stack` and any other component utilizing Flexbox use the `gap` attribute instead of `margin` for `spacing`. Safari 13 would be the only browser that doesn't support `gap`. Hopefully in the next major version, support for it can be dropped.  
My [CodeSandbox](https://codesandbox.io/s/stack-changes-proposed-iu0cn?file=/src/App.tsx) contains the modifications needed for supporting `gap` and include backward compatibility for older browsers which do not.
